src/components/UI/DateDisplay.vue

```diff
@@ -0,0 +1,92 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+type LiveStatus = 'isPlanned' | 'isStreaming' | 'isArchived'
+
+const props = defineProps<{
+  startedTime: Date
+  endedTime: Date
+}>()
+
+const status = computed((): LiveStatus => {
+  if (new Date().getTime() < props.startedTime.getTime()) return 'isPlanned'
+  if (isNaN(props.endedTime.getTime())) return 'isStreaming'
+  return 'isArchived'
+})
+
+const text = computed(() => {
+  switch (status.value) {
+    case 'isPlanned':
+      return `${
+        props.startedTime.getMonth() + 1
+      }/${props.startedTime.getDate()}`
+    case 'isStreaming':
+      return 'LIVE'
+    case 'isArchived': {
+      if (!props.endedTime) return 'LIVE'
+      const prevTime = props.endedTime.getTime()
+      const diff = new Date().getTime() - prevTime
```

Comment:
prevTime、ここでしか使っていないし命名も微妙な気がするのでdiffの定義のときに直接入れちゃっていい気がします
```suggestion
      const diff = new Date().getTime() - props.endedTime.getTime()
```